### PR TITLE
Futures uses Futures

### PR DESCRIPTION
**Pull request by som-snytt**
That's not tautological.  It uses a few more LOC, but I guess that's right.  For a couple of lines, I wonder if there is a clever monadic or combinatoric idiom.

### CODE REVIEW

File: examples/src/main/scala/futures.scala [hunk 1]
```diff
@@ -9,9 +10,20 @@ object futures {
   def g(x: Int) = x * x
 
   def main(args: Array[String]) {
-    val x = future(someLengthyComputation)
-    anotherLengthyComputation
-    val y = f(x()) + g(x())
-    println(y)
+    val d = util.Duration(1, SECONDS).toMillis
```
Comment (jsuereth) [inline examples/src/main/scala/futures.scala:13]: Just a random question.  Have you thought about showing asynchronous sequencing with Future + for-expressions?   Would make an awesome example.  If you don't have time, I'll try to pull something from a presentation I'm about to give on it.

**Comment by jsuereth** [comment]
Thanks man!

File: examples/src/main/scala/futures.scala (hunk 1 above)
Comment (som-snytt) [inline examples/src/main/scala/futures.scala:13]: No, I'm just learning API to find out how to make thread pools die.  But I'll give it thought.  I intended to contribute more here after M4. That's just a random answer.

Comment (jsuereth) [inline examples/src/main/scala/futures.scala:13]: Another random question: What's the color of a duck?

Looking forward to a truly random answer :)

Comment (som-snytt) [inline examples/src/main/scala/futures.scala:13]: Recently at Happy Hollow, we visited the "birding research station", where we had a chance to see a ring-necked teal and ask, "Is that really teal?"

My daughter's favorite part was the old typewriter on which the field researcher would type out her field notes.  I explained what a typewriter is.  She also knows what a wired phone is, and may even have seen one once.

Comment (jsuereth) [inline examples/src/main/scala/futures.scala:13]: This reminds me of the time I designed my own board game.   It was a cross of Mega Man and Monopoly.   The goal was a cross between acquiring as much stuff as possible and destroying as much stuff as you could.

**Pull request merged**